sources/api/pluto/src/main.rs

```diff
@@ -157,9 +192,62 @@ fn get_max_pods(client: &Client, session_token: &str) -> Result<String> {
     error::NoInstanceTypeMaxPods { instance_type }.fail()
 }
 
-fn get_cluster_dns_ip(client: &Client, session_token: &str) -> Result<String> {
+/// Returns the cluster's DNS IPV4 address. First it attempts to call EKS describe-cluster to find
+/// the `serviceIPv4CIDR`. If that works, it returns the first `*.10` address. If the EKS call is
+/// not successful, it falls back to using IMDS MAC CIDR blocks to return one of two default
+/// addresses.
+async fn get_cluster_dns_ip(client: &Client, session_token: &str) -> Result<String> {
+    let region = get_region().await.context(error::Region)?;
+    let cluster_name = get_cluster_name().await.context(error::ClusterName {})?;
+
+    // try calling eks describe-cluster to figure out the dns cluster ip
+    if let Some(dns_ip) = get_dns_from_eks(&region, &cluster_name).await {
+        // we were able to calculate the dns ip from the cidr range we received from eks
+        return Ok(dns_ip);
+    }
+
+    // we were unable to obtain or parse the cidr range from eks, fallback to one of two default
+    // values based on the cidr range of our primary network interface
+    get_cluster_dns_from_imds_mac(client, session_token).await
+}
+
+/// Gets the Service IPV4 CIDR setting from EKS and parses it to calculate the cluster DNS IP.
+/// Prints the error and returns `None` if anything goes wrong.
+async fn get_dns_from_eks(region: &str, cluster_name: &str) -> Option<String> {
+    let cidr = match get_cluster_cidr(region, cluster_name).await {
+        Ok(cidr) => cidr,
+        Err(e) => {
+            eprintln!("Unable to get CIDR from EKS, using default DNS IP: {}", e);
+            return None;
+        }
+    };
+    match get_dns_from_cidr(&cidr) {
+        Ok(dns_ip) => Some(dns_ip),
+        Err(e) => {
+            eprintln!("Unable to parse CIDR from EKS, using default DNS IP: {}", e);
+            None
```

Comment:
nit: since the message and error handling are the same in the two steps of this function, it seems like it might be simpler combined:
```rust
async fn get_dns_from_eks(region: &str, cluster_name: &str) -> Option<String> { 
    get_cluster_cidr(region, cluster_name).await
        .and_then(|cidr| get_dns_from_cidr(cidr))
        .map_err(|e| eprintln!("Unable to parse CIDR from EKS, using default DNS IP: {}", e))
        .ok()
}
```